src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs

```diff
@@ -510,10 +523,9 @@ private string CleanUserAgentInfoEntry(string infoEntry)
         private void MergeUserAgentInfo(List<ProductInfoHeaderValue> defaultUserAgentInfoList)
         {
             // If you want to log ProductName in userAgent, it has to be without spaces
-
             foreach(ProductInfoHeaderValue piHv in defaultUserAgentInfoList)
             {
-                if(!HttpClient.DefaultRequestHeaders.UserAgent.Any<ProductInfoHeaderValue>((hv) => hv.Product.Name.Equals(piHv.Product.Name, StringComparison.OrdinalIgnoreCase)));
+                if(!HttpClient.DefaultRequestHeaders.UserAgent.Any<ProductInfoHeaderValue>((hv) => hv.Product.Name.Equals(piHv.Product.Name, StringComparison.OrdinalIgnoreCase)))
```

Comment:
This does not account for UserAgent strings that are comments only.  They will NOT have a .Product property... this will throw a null reference exception.  If this constructor is used https://technet.microsoft.com/en-us/library/hh138008(v=vs.110) then ProductInfoHeaderValue will have only a comment.